android/src/main/java/com/amplitude/android/plugins/RemnantEventsMigrationPlugin.kt

```diff
@@ -0,0 +1,142 @@
+package com.amplitude.android.plugins
+
+import com.amplitude.android.Configuration
+import com.amplitude.android.utilities.DatabaseStorageProvider
+import com.amplitude.common.android.AndroidContextProvider
+import com.amplitude.common.android.LogcatLogger
+import com.amplitude.core.Amplitude
+import com.amplitude.core.events.BaseEvent
+import com.amplitude.core.events.Plan
+import com.amplitude.core.platform.Plugin
+import com.amplitude.core.utilities.optionalJSONObject
+import com.amplitude.core.utilities.optionalString
+import org.json.JSONArray
+import org.json.JSONObject
+import java.math.BigDecimal
+
+/**
+ * When switching the SDK from previous version to this version, remnant events might remain unsent in sqlite.
+ * This plugin:
+ *      1. reads the events from sqlite table
+ *      2. convert events to BaseEvent
+ *      3. track events with this SDK native method
+ *      4. delete events from sqlite table
+ */
+class RemnantEventsMigrationPlugin : Plugin {
+    override val type: Plugin.Type = Plugin.Type.Before
+    override lateinit var amplitude: Amplitude
+    private lateinit var contextProvider: AndroidContextProvider
+
+    override fun setup(amplitude: Amplitude) {
+        super.setup(amplitude)
+        val configuration = amplitude.configuration as Configuration
+        contextProvider =
+            AndroidContextProvider(configuration.context, configuration.locationListening)
+
+        val databaseStorage = DatabaseStorageProvider().getStorage(amplitude)
+        var maxMigratedEventId: Long = 0
+        try {
+            @Suppress("UNCHECKED_CAST")
+            val remnantEvents = databaseStorage.readEventsContent() as List<JSONObject>
```

Comment:
This is for events rights? Do you cover the identify events, and user_id, device_id, etc.?

Reply:
Yes, events, example:
```json
{
    "event_type": "marvin-test",
    "timestamp": 1656022776834,
    "user_id": "marvin",
    "device_id": "d34d76b7-ff6a-40ae-8dd3-fc93ece8fdceR",
    "session_id": 1656018188962,
    "uuid": "70b13b79-8530-4399-b39e-1aaf03ce9734",
    "sequence_number": 8,
    "version_name": "1.0",
    "os_name": "android",
    "os_version": "13",
    "api_level": 33,
    "device_brand": "google",
    "device_manufacturer": "Google",
    "device_model": "sdk_gphone64_arm64",
    "carrier": "T-Mobile",
    "country": "US",
    "language": "en",
    "platform": "Android",
    "library": {
        "name": "amplitude-android",
        "version": "2.37.1-SNAPSHOT"
    },
    "api_properties": {
        "limit_ad_tracking": false,
        "gps_enabled": true
    },
    "event_properties": {},
    "user_properties": {},
    "groups": {},
    "group_properties": {}
}
```
`user_id` and `device_id` are already embedded. I am not sure about `identify events` you mentioned, are they special events? (I didn't find any related desc in https://developers.amplitude.com/docs/android, or in my local db). Do you know how to generate those if they are not the events I posted above? Once I figure out how to generate it and confirm we need it, I can update to try to cover those too.

Reply:
- get `user_id`, `device_id`, `session_id`, `opt_out`...
- get `identify` events